token-metadata/program/src/processor/metadata/transfer.rs

```diff
@@ -142,49 +167,36 @@ fn transfer_v1(program_id: &Pubkey, ctx: Context<Transfer>, args: TransferArgs)
         token_program: ctx.accounts.spl_token_program_info.clone(),
     };
 
-    let auth_rules_validate_params = AuthRulesValidateParams {
-        mint_info: ctx.accounts.mint_info,
-        target_info: Some(ctx.accounts.destination_owner_info),
-        authority_info: Some(ctx.accounts.authority_info),
-        owner_info: None,
-        programmable_config: metadata.programmable_config.clone(),
-        amount,
-        auth_data,
-        auth_rules_info: ctx.accounts.authorization_rules_info,
-        operation: Operation::Transfer,
-    };
-
     let token_standard = metadata
         .token_standard
         .ok_or(MetadataError::InvalidTokenStandard)?;
 
+    // Wallet-to-wallet transfer are always allowed except if a sale
+    // delegate is set.
+    let is_wallet_to_wallet = ctx.accounts.token_owner_info.owner == &system_program::ID
+        && ctx.accounts.destination_owner_info.owner == &system_program::ID;
+
     // Sale delegates prevent any other kind of transfer.
     let is_sale_delegate_set = if let Some(delegate_role) = metadata.persistent_delegate {
         delegate_role == DelegateRole::Sale
     } else {
         false
     };
 
-    // Wallet-to-wallet transfer are always allowed except if a sale
-    // delegate is set.
-    let is_wallet_to_wallet = ctx.accounts.token_owner_info.owner == &system_program::ID
-        && ctx.accounts.destination_owner_info.owner == &system_program::ID;
-
     // Determine transfer type.
     let transfer_authority = if let Some(delegate_role) = metadata.persistent_delegate {
```

Comment:
There is now a "generic" function to determine the authority of a request `AuthorityType::get_authority_type`. This will check whether it is the `Holder` (token owner), `Delegate` or `Metadata` (update authority). Wonder if this could/should be used here so we have a consistent way to determine the authority.